From c8e0a0417b927fc81ea42c5f05a4aa6fbcbf1533 Mon Sep 17 00:00:00 2001
From: Steve Kondik <shade@chemlab.org>
Date: Fri, 23 Aug 2013 21:12:12 -0700
Subject: [PATCH 15/16] sdcard: Allow building as a static library

 * Split main() out in order to allow linking with a multi-call
   executable such as recovery.

Change-Id: I3325135e47e4dd5af3912f5ce67ec64007d54a63
---
 sdcard/Android.bp | 21 +++++++++++++++++++--
 sdcard/main.c     | 21 +++++++++++++++++++++
 sdcard/sdcard.cpp |  2 +-
 3 files changed, 41 insertions(+), 3 deletions(-)
 create mode 100644 sdcard/main.c

diff --git a/sdcard/Android.bp b/sdcard/Android.bp
index c096587..343ab79 100644
--- a/sdcard/Android.bp
+++ b/sdcard/Android.bp
@@ -1,6 +1,6 @@
-cc_binary {
+cc_library_static {
     srcs: ["sdcard.cpp"],
-    name: "sdcard",
+    name: "libsdcard",
     cflags: [
         "-Wall",
         "-Wno-unused-parameter",
@@ -15,3 +15,20 @@ cc_binary {
         misc_undefined: ["integer"],
     },
 }
+
+cc_binary {
+    srcs: ["main.c"],
+    name: "sdcard",
+    cflags: [
+        "-Wall",
+        "-Wno-unused-parameter",
+        "-Werror",
+    ],
+    static_libs: ["libsdcard"],
+    shared_libs: [
+        "libbase",
+        "libc",
+        "libcutils",
+        "libminijail",
+    ],
+}
diff --git a/sdcard/main.c b/sdcard/main.c
new file mode 100644
index 0000000..ad2405b
--- /dev/null
+++ b/sdcard/main.c
@@ -0,0 +1,21 @@
+/*
+ * Copyright (C) 2010 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+extern int sdcard_main(int argc, char **argv);
+
+int main(int argc, char **argv) {
+    return sdcard_main(argc, argv);
+}
diff --git a/sdcard/sdcard.cpp b/sdcard/sdcard.cpp
index 2b35819..0cc3e10 100644
--- a/sdcard/sdcard.cpp
+++ b/sdcard/sdcard.cpp
@@ -232,7 +232,7 @@ static int usage() {
     return 1;
 }
 
-int main(int argc, char **argv) {
+extern "C" int sdcard_main(int argc, char **argv) {
     const char *source_path = NULL;
     const char *label = NULL;
     uid_t uid = 0;
-- 
2.7.4

